### PR TITLE
Fix pickling error on network file system

### PR DESCRIPTION
**Pull request by misterbrandonwalker**
To summarize without the additions to catch errors when files are inaccessible or missing on a network file system, toil will throw this stack trace below.

```
        Traceback (most recent call last):
          File "/home/walkerbd/toil/src/toil/worker.py", line 390, in workerScript
            with fileStore.open(job):
          File "/home/walkerbd/mambaforge-pypy3/envs/dali/lib/python3.9/contextlib.py", line 119, in __enter__
            return next(self.gen)
          File "/home/walkerbd/toil/src/toil/fileStores/nonCachingFileStore.py", line 105, in open
            self._removeDeadJobs(self.coordination_dir)
          File "/home/walkerbd/toil/src/toil/fileStores/nonCachingFileStore.py", line 243, in _removeDeadJobs
            for jobState in cls._getAllJobStates(coordination_dir):
          File "/home/walkerbd/toil/src/toil/fileStores/nonCachingFileStore.py", line 300, in _getAllJobStates
            yield NonCachingFileStore._readJobState(fname)
          File "/home/walkerbd/toil/src/toil/lib/retry.py", line 292, in call
            return func(*args, **kwargs)
          File "/home/walkerbd/toil/src/toil/fileStores/nonCachingFileStore.py", line 319, in _readJobState
            state = dill.load(fH)
          File "/home/walkerbd/mambaforge-pypy3/envs/dali/lib/python3.9/site-packages/dill/_dill.py", line 287, in load
            return Unpickler(file, ignore=ignore, **kwds).load()
          File "/home/walkerbd/mambaforge-pypy3/envs/dali/lib/python3.9/site-packages/dill/_dill.py", line 442, in load
            obj = StockUnpickler.load(self)
        OSError: [Errno 5] Input/output error
```

## Changelog Entry
To be copied to the [draft changelog](https://github.com/DataBiosphere/toil/wiki/Draft-Changelog) by merger:

 * PR submitter writes their recommendation for a changelog entry here

## Reviewer Checklist

 * [ ] Make sure it is coming from `issues/XXXX-fix-the-thing` in the Toil repo, or from an external repo.
    * [ ] If it is coming from an external repo, make sure to pull it in for CI with:
        ```
        contrib/admin/test-pr otheruser theirbranchname issues/XXXX-fix-the-thing
        ```
    * [ ] If there is no associated issue, [create one](https://github.com/DataBiosphere/toil/issues/new).
* [ ] Read through the code changes. Make sure that it doesn't have:
    * [ ] Addition of trailing whitespace.
    * [ ] New variable or member names in `camelCase` that want to be in `snake_case`.
    * [ ] New functions without [type hints](https://docs.python.org/3/library/typing.html).
    * [ ] New functions or classes without informative docstrings.
    * [ ] Changes to semantics not reflected in the relevant docstrings.
    * [ ] New or changed command line options for Toil workflows that are not reflected in `docs/running/{cliOptions,cwl,wdl}.rst` 
    * [ ] New features without tests.
* [ ] Comment on the lines of code where problems exist with a review comment. You can shift-click the line numbers in the diff to select multiple lines.
* [ ] Finish the review with an overall description of your opinion.

## Merger Checklist

* [ ] Make sure the PR passes tests.
* [ ] Make sure the PR has been reviewed **since its last modification**. If not, review it.
* [ ] Merge with the Github "Squash and merge" feature.
    * [ ] If there are multiple authors' commits, add [Co-authored-by](https://github.blog/2018-01-29-commit-together-with-co-authors/) to give credit to all contributing authors.
* [ ] Copy its recommended changelog entry to the [Draft Changelog](https://github.com/DataBiosphere/toil/wiki/Draft-Changelog).
* [ ] Append the issue number in parentheses to the changelog entry.

### CODE REVIEW

File: src/toil/lib/threading.py [hunk 1]
```diff
@@ -533,6 +533,9 @@ def leave(self) -> Iterator[bool]:
                 # There is someone claiming to be here. Are they alive?
                 full_path = os.path.join(self.lockfileDir, item)
 
+                if not os.path.exists(full_path): # suddenly file doesnt exist on network file system?
+                    continue
+
```
Comment (adamnovak) [inline src/toil/lib/threading.py:538]: This isn't going to 100% close the race here between opening the file and someone deleting it. We probably want to try/catch around the open instead.

**Review by adamnovak — COMMENTED** [review]
This looks good but I think I want to actually handle open failures in the arena leave.

File: src/toil/lib/threading.py (hunk 1 above)
Comment (misterbrandonwalker) [reply src/toil/lib/threading.py]: Thanks for getting back to me so fast and excellent point! How does it look now?

Comment (misterbrandonwalker) [reply src/toil/lib/threading.py]: Hey @adamnovak I hope you are well. Was hoping could get this snippet reviewed and then merged? We love toil and are using it as our workflow engine!

Comment (adamnovak) [reply src/toil/lib/threading.py]: Oh sorry, I didn't notice this was updated!

**Review by adamnovak — APPROVED** [review]

**Comment by adamnovak** [comment]
I have pulled this in to the `issues/4575-tolerate-io-error` branch to run tests. If it passes we can merge it.

**Pull request merged**